File: android/src/main/kotlin/com/mytiki/sdk/capture/receipt/capacitor/plugin/js/JSRetailer.kt
/*
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in the root directory.
 */

package com.mytiki.sdk.capture.receipt.capacitor.plugin.js

import com.getcapacitor.JSObject
import com.microblink.core.Retailer

/**
 * Represents a Retailer for the Receipt Scanning Plugin (RSP).
 *
 * @param retailer The retailer object to be wrapped by JSRetailer.
 */
class JSRetailer(retailer: Retailer) {
    private val id: Int
    private val bannerId: Int

    init {
        id = retailer.id()
        bannerId = retailer.bannerId()
    }

    /**
     * Converts the JSRetailer object to a JSON representation.
     *
     * @return A JSON object representing the JSRetailer.
     */
    fun toJS(): JSObject =
        JSObject()
            .put("id", id)
            .put("bannerId", bannerId)

    companion object {
        /**
         * Create an optional JSRetailer object from a Retailer object.
         *
         * @param retailer The Retailer object to be converted.
         * @return An JSRetailer object if the input Retailer is not null, otherwise null.
         */
        fun opt(retailer: Retailer?): JSRetailer? =
            if (retailer != null) JSRetailer(retailer) else null
    }
}

File: ios/Plugin/Plugin/Js/JSRetailer.swift
/*
 * RspStringType Struct
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in root directory.
 */

import Foundation
import BlinkReceipt
import BlinkEReceipt
import Capacitor

/**
 Represents a response containing retailer identification information

 This struct is used to convey a retailer id value along with its banner id.
 */
struct JSRetailer{
    /// The retailer's id
    private let id: Int
    /// The retailer's banner id
    private let bannerId: Int?

    /**
     Initializes an `JSStringType` struct.

     - Parameters:
        - retailer: The retailer value.
     */
    init(retailer: WFRetailerId) {
        id = Int(retailer.rawValue)
        bannerId = nil
    }

    /**
     Converts the `JSStringType` struct into a dictionary suitable for use in plugin response data.

     - Returns: A dictionary containing the id value and its banner id in a format suitable for a Capacitor plugin call result.
     */
    public func toJSObject() -> JSObject {
        var ret = JSObject()
        ret["id"] = id
        ret["bannerId"] = bannerId
        return ret
    }
}
